src/googleError.ts

```diff
@@ -14,9 +14,64 @@
  * limitations under the License.
  */
 
-import {Status} from './status';
-
-export class GoogleError extends Error {
-  code?: Status;
-  note?: string;
-}
+ import {Status} from './status';
+ import * as protobuf from 'protobufjs';
+ import { Metadata } from '@google-cloud/common';
+ export class GoogleError extends Error {
+   code?: Status;
+   note?: string;
+   metadata?: Metadata;
+   statusDetails?: protobuf.Message<{}> | protobuf.Message<{}>[];
+ }
+  
+ interface ProtobufAny {
+   type_url: string;
+   value: Uint8Array;
+ }
+  
+ interface RpcStatus {
+   code: number;
+   message: string;
+   details: ProtobufAny[];
+ }
+  
+ export class GoogleErrorDecoder {
+   root: protobuf.Root;
+   anyType: protobuf.Type;
+   statusType: protobuf.Type;
+  
+   constructor() {
+     // eslint-disable-next-line @typescript-eslint/no-var-requires
+     const errorProtoJson = require('../../protos/status.json');
+     this.root = protobuf.Root.fromJSON(errorProtoJson);
+     this.anyType = this.root.lookupType('google.protobuf.Any');
+     this.statusType = this.root.lookupType('google.rpc.Status');
+   }
+  
+   decodeProtobufAny(anyValue: ProtobufAny): protobuf.Message<{}> {
```

Comment:
This code already exists in `src/fallbackError.ts`, can we just import the required functions from there instead of duplicating?  We could rename `src/fallbackError.ts` to `src/googleError.ts` since it's now not about the fallback (one of the modes), but more like a general error handling code.